### PR TITLE
Add PSPNet

### PR DESCRIPTION
**Pull request by zhanghang1989**

### CODE REVIEW

**Comment by mli** [comment]
Job PR-179-4 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-179/4/index.html 
Code coverage of this PR:  vs. Master:

**Comment by mli** [comment]
Job PR-179-6 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-179/6/index.html 
Code coverage of this PR:  vs. Master:

File: gluoncv/model_zoo/syncbn.py [hunk 1]
```diff
@@ -1,13 +1,111 @@
...
+from mxnet import gluon, autograd, test_utils
 
-import numpy as np
+class SharedTensor(object):
+    """Shared Tensor for Syncing"""
+    def __init__(self, key, nchannels, nGPUs):
+        self.mutex = threading.Lock()
+        self.all_tasks_done = threading.Condition(self.mutex)
+        self._key = key
+        self.nGPUs = int(nGPUs)
```
Comment (zhreshold) [inline gluoncv/model_zoo/syncbn.py]: name convention should be consistent, no camel case for variables

File: gluoncv/model_zoo/syncbn.py [hunk 2]
```diff
@@ -1,13 +1,111 @@
...
-from mxnet.gluon import HybridBlock
+import mxnet as mx
+from mxnet import gluon, autograd, test_utils
 
-import numpy as np
+class SharedTensor(object):
+    """Shared Tensor for Syncing"""
+    def __init__(self, key, nchannels, nGPUs):
+        self.mutex = threading.Lock()
+        self.all_tasks_done = threading.Condition(self.mutex)
```
Comment (zhreshold) [inline gluoncv/model_zoo/syncbn.py]: is this private? self._all_tasks_done

File: gluoncv/model_zoo/syncbn.py [hunk 3]
```diff
@@ -1,13 +1,111 @@
...
+        self.out = mx.nd.zeros(nchannels)
+        self._clear()
+
+    def _clear(self):
+        self.list = []
+        self.push_tasks = self.nGPUs
+        self.reduce_tasks = self.nGPUs
+
+    def push(self, t):
+        """push to _SharedTensor"""
```
Comment (zhreshold) [inline gluoncv/model_zoo/syncbn.py]: docstring is not readable

File: gluoncv/model_zoo/syncbn.py [hunk 4]
```diff
@@ -1,13 +1,111 @@
...
-from mxnet import autograd, test_utils
-from mxnet.gluon import HybridBlock
+import mxnet as mx
+from mxnet import gluon, autograd, test_utils
 
-import numpy as np
+class SharedTensor(object):
+    """Shared Tensor for Syncing"""
+    def __init__(self, key, nchannels, nGPUs):
+        self.mutex = threading.Lock()
```
Comment (zhreshold) [inline gluoncv/model_zoo/syncbn.py]: mutex should be private

**Comment by zhreshold** [comment]
train_psp tutorial took 2min for each iteration, therefore overall 6min+ per tutorial is not acceptable. 
The same applies to train_fcn tutorial, we should try to limit each tutorial within 1min or 2min.

**Comment by mli** [comment]
Job PR-179-7 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-179/7/index.html 
Code coverage of this PR:  vs. Master:

**Comment by zhanghang1989** [comment]
@zhreshold The train tutorial is slow due to using cpu. When using GPU, the CI often crashes. I am reducing the number of iters in the tutorial to save time.

**Pull request merged**